### ISSUE TITLE
Issue with CICD step "Deploy" for multi-project repository 

### ISSUE THREAD

**Issue by gntpet** [microsoft/AL-Go#194]
Hey @freddydk,
I got another case with multi-project repository. There's similar issue with CICD deployment step like in the issue #193 "Publish To Environment".

When CICD runs for multi-project repository, it deploys all artifacts to all environments, while ignoring **project** setting.

I believe, it's related to the fact that we are passing artifacts folder as a parameter. 

![image](https://user-images.githubusercontent.com/16342128/189066818-dd641eb9-e5e0-4c8b-bb40-4476bbd9e168.png)

code just simply reads all apps from that folder. Thus totally ignoring projects param.
![image](https://user-images.githubusercontent.com/16342128/189067332-9a5e2686-10a1-4ab3-afee-21b32a12fada.png)

**Comment by freddydk** [comment]
Will fix this

**Comment by freddydk** [comment]
fixed in the latest preview

**Comment by gntpet** [comment]
hey @freddydk, 
I just updated to the latest preview, and tested CICD pipeline. Now, it does not deploy artifacts at all. It fails with an error

```
--
Deploy action failed. Error: There is no artifacts present in D:\a\srs-bc\srs-bc\artifacts. Stacktrace: at <ScriptBlock>, D:\a\_actions\microsoft\AL-Go-Actions\preview\Deploy\Deploy.ps1: line 47 at <ScriptBlock>, D:\a\_actions\microsoft\AL-Go-Actions\preview\Deploy\Deploy.ps1: line 42 at <ScriptBlock>, D:\a\_temp\9c23d17f-84dc-45a8-a949-f7fc6aaa24e0.ps1: line 1 at <ScriptBlock>, <No file>: line 1Show less

<br class="Apple-interchange-newline">
```

<img width="1295" alt="image" src="https://user-images.githubusercontent.com/16342128/189612852-39c9d104-56cd-4596-8dcd-6dc1c4032079.png">

**Comment by freddydk** [comment]
Strange - and sorry about that.
Any chance I can see the full deploy output?

**Comment by freddydk** [comment]
Got it, found my mistake

**Comment by freddydk** [comment]
Issue was that when using the projects secret, it wasn't decoded correctly.
When using the new \<environment\>_projects setting, it worked.
Both of course needs to work:-)

**Comment by gntpet** [comment]
hey @freddydk, 
i can confirm it is working correctly now

**Comment by freddydk** [comment]
FYI - I did another update of preview today, with some optimizations and better support for UTF8.

**Comment by gntpet** [comment]
thanks, will give a try

**Comment by freddydk** [comment]
Shipped in v2.0
